### ISSUE TITLE
Error building v0.24.4-rc1on Android

### ISSUE THREAD

**Issue by swansontec** [libgit2/libgit2#3635]
This build failure occurs under the Android NDK for v0.24.4-rc1. This is a regression since version, 0.23.4 which compiled without errors.

```
libgit2-0.24.0-rc1/src/fileops.c: In function 'git_futils_filestamp_check':
libgit2-0.24.0-rc1/src/fileops.c:1037:40: error: 'struct stat' has no member named 'st_mtim'
  const struct timespec *statmtime = &st.st_mtim;
                                        ^
libgit2-0.24.0-rc1/src/fileops.c: In function 'git_futils_filestamp_set_from_stat':
libgit2-0.24.0-rc1/src/fileops.c:1079:40: error: 'struct stat' has no member named 'st_mtim'
  const struct timespec *statmtime = &st->st_mtim;
```

**Comment by carlosmn** [comment]
We switched over to do auto-detection, so we shouldn't need to care about the particular OS. This is presumably a cross-compilation, right? Are we not doing the right thing there?

**Comment by ethomson** [comment]
We're not - in this case we're assuming that a `struct stat` has a `struct timespec`.  Android's does not:

```
    unsigned long       st_atime;
    unsigned long       st_atime_nsec;
    unsigned long       st_mtime;
    unsigned long       st_mtime_nsec;
```

Which is despicable.  But whatever.  Patch forthcoming.

**Comment by swansontec** [comment]
Thanks.

**Comment by ethomson** [comment]
@swansontec No problem, thanks for the helpful report.

**Comment by carlosmn** [comment]
Fixed via #3638
